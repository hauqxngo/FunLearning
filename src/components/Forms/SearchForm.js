import React, { useState } from "react";
import { Link } from "react-router-dom";
import { Form, FormGroup, Label, Input, Button } from "reactstrap";

/** Search box in Category List page
 *
 * CategoryList -> SearchForm
 */

const SearchForm = ({ searchFor }) => {
  const [searchTerm, setSearchTerm] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();
    // get rid of the extra spaces
    let term = searchTerm.trim();
    searchFor(term || undefined);
    setSearchTerm("");
  };

  // update value in form
  const handleChange = (e) => {
    setSearchTerm(e.target.value);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <FormGroup className="SearchForm mx-5">
        <Label for="searchForm"></Label>
        <Input
          id="searchForm"
          name="search"
          placeholder="Search by name..."
          value={searchTerm}
          onChange={handleChange}
        />
        <div className="col text-center my-2">
          <Button type="submit" color="success">
            Search
          </Button>
          <Button className="mx-1" color="success" outline>
            <Link to="/categories" />
            Reset
          </Button>
        </div>
      </FormGroup>
    </Form>
  );
};

export default SearchForm;
