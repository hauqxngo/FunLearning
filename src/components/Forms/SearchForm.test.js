import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "./SearchForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <SearchForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
