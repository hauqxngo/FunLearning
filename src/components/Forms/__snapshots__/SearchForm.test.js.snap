// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="SearchForm mx-5 mb-3"
    >
      <label
        class="form-label"
        for="searchForm"
      />
      <input
        class="form-control"
        id="searchForm"
        name="search"
        placeholder="Search by name..."
        type="text"
        value=""
      />
      <div
        class="col text-center my-2"
      >
        <button
          class="btn btn-success"
          type="submit"
        >
          Search
        </button>
        <button
          class="mx-1 btn btn-outline-success"
        >
          <a
            href="/categories"
          />
          Reset
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
